Pod/Classes/VOKCoreDataManager.h

```diff
@@ -243,25 +245,27 @@ typedef void(^VOKObjectIDsReturnBlock)(VOKArrayOfManagedObjectIDs *managedObject
                         context:(nullable NSManagedObjectContext *)contextOrNil;
 
 /**
- Saves the main context asynchronously on the main queue. If already on the main queue it will block and save synchronously.
+ Saves the main context asynchronously.
  */
 - (void)saveMainContext;
 
 /**
- Saves the main context synchronously on the main queue. If already on the main queue it performs the same as saveMainContext.
+ Saves the main context synchronously. This method will not return until the main context is saved.
  */
 - (void)saveMainContextAndWait;
 
 /**
- Provides a managed object context for scratch work or background processing using the same persistent store coordinator as the main context. As with all managed object contexts, it is not thread-safe.
+ Provides a managed object context for scratch work or background processing as a child of the main context. As with all managed object contexts, it is not thread-safe.
```

Comment:
If you save a child context, the changes only get saved up to the immediate parent context.  The parent context changes do not automatically get saved too.  (That is the reason for the recursive `saveContextToRoot:` method.)